corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -581,6 +582,17 @@ def soft_undelete_forms(domain, form_ids):
 
         return return_value
 
+    @staticmethod
+    def modify_attachment_xml_and_metadata(form_data, form_attachment_new_xml):
+        attachment_metadata = form_data.get_attachment_meta("form.xml")
+        # Write the new xml to the database
+        XFormAttachmentSQL.write_content(attachment_metadata, form_attachment_new_xml)
+        attachment_metadata.save()
+        operation = XFormOperationSQL(user_id=SYSTEM_USER_ID, date=datetime.utcnow(),
+                                      operation='Scrub username for GDPR compliance.')
+        form_data.history.append(operation)  # TODO: should this show in Form History tab? it doesn't
```

Comment:
this won't work for SQL. Rather do this:
```python
form_data.track_create(operation)
FormAccessorSQL.update_form(form_data)
```

Reply:
Updated here (and did some extra test stuff which I clean up in the below pr):
https://github.com/dimagi/commcare-hq/pull/19979/commits/c38433c08998c958cf9e44cdd6f5a26248314aa8

And tested here:
https://github.com/dimagi/commcare-hq/pull/19979/commits/8f4cf33e4b0319561c0110c5bb470e9790b3bb27